react/ProductListContext.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useContext, useReducer, FC } from 'react'
+
+interface State {
+  visibleProducts: any[]
```

Comment:
Wouldn't it be better to make this a bit more generic?
Or even add a new property to the State called just `products`?

Reply:
It was called `products` but then @klzns asked to change it so its purpose is more clear. I like the idea of adding a new property on the state called `products`, but since it wouldn't be used now, I am not sure if it is a good idea to add this prop on this PR